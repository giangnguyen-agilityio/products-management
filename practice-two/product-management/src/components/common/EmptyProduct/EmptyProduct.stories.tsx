import type { Meta, StoryObj } from '@storybook/react'
import EmptyProduct from './index'

const meta = {
  title: 'Practice Two/EmptyProduct Component',
  component: EmptyProduct,
  decorators: [
    Story => (
      <div
        style={{
          color: '#ECB203',
        }}
      >
        <Story />
      </div>
    ),
  ],
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof EmptyProduct>

export default meta
type Story = StoryObj<typeof meta>

export const Primary: Story = {
  args: {
    errorMessage: "We couldn't find any product",
  },
}
