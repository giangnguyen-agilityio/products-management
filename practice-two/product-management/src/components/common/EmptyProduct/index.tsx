import React, { memo } from 'react'
import errorImage from '@assets/images/error-image.webp'
import { Flex, Text, Box, Image } from '@chakra-ui/react'

// Define the props interface for the EmptyProduct component
interface EmptyProductProps {
  errorMessage?: string
}

const EmptyProduct: React.FC<EmptyProductProps> = props => {
  const { errorMessage } = props

  return (
    // Container for the empty product message
    <Flex
      className="empty-product"
      flexDirection="column"
      alignItems="center"
      justifyContent="center"
      background="background"
      maxWidth={{
        sm: '340px',
        md: '720px',
        lg: '940px',
        xl: '1140px',
      }}
      margin="40px auto"
      borderRadius={6}
    >
      {/* Box containing the error image */}
      <Box className="error-image">
        {/* Display the error image */}
        <Image width="full" src={errorImage} alt="This is the error image" />
      </Box>
      {/* Text displaying the error message */}
      <Text
        className="error-title"
        color="secondary"
        fontSize={{ base: '16px', md: '20px', lg: '24px', xl: '30px' }}
        fontFamily="Oswald-Regular"
        paddingBottom={{ base: 2, md: 4, lg: 6 }}
      >
        {/* Display the error message passed as prop */}
        {errorMessage}
      </Text>
    </Flex>
  )
}

export default memo(EmptyProduct)
