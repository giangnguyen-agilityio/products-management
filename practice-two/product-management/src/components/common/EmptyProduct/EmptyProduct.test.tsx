import { render } from '@testing-library/react'
import EmptyProduct from './'

describe('EmptyProduct Component', () => {
  jest.mock('@assets/images/error-image.webp', () => ({
    default: 'errorImage',
  }))

  it('renders without error message', () => {
    const { container, getByAltText } = render(<EmptyProduct />)

    const errorImageElement = getByAltText('This is the error image')

    expect(errorImageElement).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('renders with error message', () => {
    const errorMessage = 'Sample Error Message'
    const { container, getByAltText, getByText } = render(
      <EmptyProduct errorMessage={errorMessage} />
    )

    const errorImageElement = getByAltText('This is the error image')
    const errorTitleElement = getByText(errorMessage)

    expect(errorImageElement).toBeInTheDocument()
    expect(errorTitleElement).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
