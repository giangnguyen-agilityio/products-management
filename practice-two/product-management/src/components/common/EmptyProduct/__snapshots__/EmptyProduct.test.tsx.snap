// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyProduct Component renders with error message 1`] = `
<div>
  <section
    class="chakra-container empty-product css-0"
  >
    <div
      class="css-28uq1i"
    >
      <div
        class="error-image css-0"
      >
        <img
          alt="This is the error image"
          class="chakra-image css-13rbvqk"
          src=""
        />
      </div>
      <p
        class="chakra-text error-title css-1u3xqgt"
      >
        Sample Error Message
      </p>
    </div>
  </section>
</div>
`;

exports[`EmptyProduct Component renders without error message 1`] = `
<div>
  <section
    class="chakra-container empty-product css-0"
  >
    <div
      class="css-28uq1i"
    >
      <div
        class="error-image css-0"
      >
        <img
          alt="This is the error image"
          class="chakra-image css-13rbvqk"
          src=""
        />
      </div>
      <p
        class="chakra-text error-title css-1u3xqgt"
      />
    </div>
  </section>
</div>
`;
